app/javascript/vue/books-data.js

```diff
@@ -1,7 +1,27 @@
 import axios from 'axios'
+import _ from 'lodash'
 
-export default new Promise((resolve, reject) => {
-  axios('/api/books')
-    .then((response) => resolve(response.data))
-    .catch((e) => reject(e))
-})
+function getAll () {
+  return new Promise((resolve, reject) => {
```

Comment:
Axios already returns promise so this method can be simplified to:

```
return axios('/api/books')
```

And after that you just do something like:

```
booksData.getAll().then((response) => { yourBooks = response.data }).catch(...)
```

Reply:
Let's discuss more

We have the promise and it takes care about retrieving books
So, it's the easiest export to use it outside

Why do we need to know about the response structure?
I suggest to keep this version

Reply:
@resivalex just not sure why we should wrap it in another promise if axios returns promise we check for response/error in component already by using then/catch. Thoughts on that?

Reply:
Minus: extra wrap
Pros: a higher level of abstraction (Imaginary: it can be replaced with any promise which returns just books

Reply:
@resivalex do you mean simplification or response data here as an advantage? Sorry just trying to follow your thought! Maybe an example from you will help?

Reply:
Of cause,

For example, a user need to select a set of books (e.g. ShareBooksSelector). We can wrap this process in Promise, which returns an array of books and use our BookShelf without any problems

Reply:
@resivalex ok I think I see your point! Basically we simplify return value so we don't have to call `.data` on the response! We can keep it as it is! Thanks for explanation!